src/components/ColorPicker/ColorPicker.tsx

```diff
@@ -0,0 +1,333 @@
+                    <div
+                        className={merge([
+                            "tw-flex-1 tw-grid tw-gap-2 tw-justify-items-stretch",
+                            colorFormat === ColorFormat.Hex ? "tw-grid-flow-col" : "tw-grid-flow-row",
+                        ])}
+                    >
+                        {colorFormat === ColorFormat.Hex ? (
+                            <TextInput
+                                defaultValue={hex.substring(1)}
+                                decorator="#"
+                                onInput={(hex) => isValidHex(hex) && onSelect(toColor(currentColor, { hex }))}
+                            />
+                        ) : (
+                            <>
+                                <TextInput
+                                    min={0}
+                                    max={255}
+                                    type={TextInputType.Number}
+                                    defaultValue={`${rgb.r}`}
+                                    decorator="R"
+                                    onInput={(r) => onSelect(toColor(currentColor, { rgba: { r } }))}
+                                />
+                                <TextInput
+                                    min={0}
+                                    max={255}
+                                    type={TextInputType.Number}
+                                    defaultValue={`${rgb.g}`}
+                                    decorator="G"
+                                    onInput={(g) => onSelect(toColor(currentColor, { rgba: { g } }))}
+                                />
+                                <TextInput
+                                    min={0}
+                                    max={255}
+                                    type={TextInputType.Number}
+                                    defaultValue={`${rgb.b}`}
+                                    decorator="B"
+                                    onInput={(b) => onSelect(toColor(currentColor, { rgba: { b } }))}
+                                />
+                            </>
+                        )}
+                    </div>
+                    <div className="tw-w-24">
+                        <TextInput
+                            min={0}
+                            max={100}
+                            type={TextInputType.Number}
+                            defaultValue={`${rgb.a * 100}`}
+                            decorator="%"
+                            onInput={(a) =>
+                                onSelect(toColor(currentColor, { rgba: { a: parseInt(a || "0", 10) / 100 } }))
+                            }
+                        />
+                    </div>
+                </div>
+            </div>
+        </div>
+    );
+};
+
+enum BrandColorView {
+    List = "List",
+    Grid = "Grid",
+}
+const BrandColorPicker: FC<ColorPickerProps> = ({ palettes: defaultPalettes = [], currentColor, onSelect }) => {
```

Comment:
I'd create separate files for the `BrandColorPicker` and `CustomColorPicker` components to improve readability, wdyt?

Reply:
Just started looking into the component and it wasn't really an issue, so I'd say we'll stick what we have :)